discovery/azure/azure.go

```diff
@@ -597,35 +627,19 @@ var errorNotFound = errors.New("network interface does not exist")
 // getNetworkInterfaceByID gets the network interface.
 // If a 404 is returned from the Azure API, `errorNotFound` is returned.
 // On all other errors, an autorest.DetailedError is returned.
-func (client *azureClient) getNetworkInterfaceByID(ctx context.Context, networkInterfaceID string) (*network.Interface, error) {
-	result := network.Interface{}
-	queryParameters := map[string]interface{}{
-		"api-version": "2018-10-01",
-	}
-
-	preparer := autorest.CreatePreparer(
-		autorest.AsGet(),
-		autorest.WithBaseURL(client.nic.BaseURI),
-		autorest.WithPath(networkInterfaceID),
-		autorest.WithQueryParameters(queryParameters),
-		autorest.WithUserAgent(userAgent))
-	req, err := preparer.Prepare((&http.Request{}).WithContext(ctx))
-	if err != nil {
-		return nil, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", nil, "Failure preparing request")
-	}
-
-	resp, err := client.nic.GetSender(req)
-	if err != nil {
-		return nil, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure sending request")
-	}
+func (client *azureClient) getNetworkInterfaceByID(ctx context.Context, networkInterfaceID string) (*armnetwork.Interface, error) {
+	splitedNetworkInterfaceID := strings.Split(networkInterfaceID, "/")
+	resourceGroupName := splitedNetworkInterfaceID[4]
+	networkInterfaceName := splitedNetworkInterfaceID[8]
 
-	result, err = client.nic.GetResponder(resp)
+	resp, err := client.nic.Get(ctx, resourceGroupName, networkInterfaceName, nil)
 	if err != nil {
-		if resp.StatusCode == http.StatusNotFound {
+		var responseError *azcore.ResponseError
+		if errors.As(err, &responseError) && responseError.StatusCode == http.StatusNotFound {
 			return nil, errorNotFound
 		}
-		return nil, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
+		return nil, autorest.NewErrorWithError(err, "armnetwork.InterfacesClient", "Get", nil, "Failure responding to request")
```

Comment:
What would be the best way to get rid of autorest completly ?

Reply:
I think it depends on your logic about error, rely on autrest to wrap error is not necessary. Our new version of SDK does not provide utility about errors, mainly because the error returned by SDK is simple.